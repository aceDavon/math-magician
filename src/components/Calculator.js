import * as React from 'react';
import { operands, Symbols } from './Data';
import calculate from '../logic/Calculate';

const Calculator = () => {
  const [obj, setData] = React.useState({
    total: '',
    next: null,
    operation: null,
  });

  const { total, next, operation } = obj;
  const handleClick = (e) => {
    const data = e.target.value;
    const state = calculate(obj, data);
    setData(state);
  };

  return (
    <div className="w-1/3 rounded-lg shadow-lg shadow-gray-200 py-4">
      <div className="flex flex-col gap-0 w-9/12 mx-auto">
        <div className="bg-slate-500 w-full text-white text-right h-10 flex items-center px-4">
          {total}
          {operation}
          {next}
        </div>
        <div className="flex gap-0 w-full">
          <div className="bg-gray-300 grid grid-cols-3 w-full">
            {operands.map((button) => (
              <button
                type="button"
                className="bg-gray-300 border px-2 py-2"
                value={button}
                key={button}
                onClick={handleClick}
              >
                {button}
              </button>
            ))}
            <button
              type="button"
              value={0}
              className="col col-span-2 border bg-gray-300 py-2"
              onClick={handleClick}
            >
              0
            </button>
            <button
              type="button"
              value="."
              className="px-2 py-2 border bg-gray-300"
              onClick={handleClick}
            >
              .
            </button>
          </div>
          <div className="w-3/12 flex flex-col">
            {Symbols.map((signs) => (
              <button
                type="button"
                className="bg-orange-400 border px-2 py-2"
                key={signs}
                onClick={handleClick}
                value={signs}
              >
                {signs}
              </button>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
