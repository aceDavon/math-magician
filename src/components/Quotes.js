import React from 'react';

const Quotes = () => {
  const [arr, setArr] = React.useState({
    id: 0,
    author: 'Anonymous',
    quote: 'Math is the source code of reality.',
  });

  const fetchData = async () => {
    const data = await fetch('https://random-math-quote-api.herokuapp.com/');

    const res = await data.json();

    setArr(res);
  };

  const { quote, author } = arr;

  return (
    <div className="w-full pt-20 relative h-screen bg-gradient-to-r from-gray-300 to-gray-500 text-black">
      <div className="mt-28 w-4/12 mx-auto px-8 py-4 text-center shadow-sm shadow-gray-100 bg-white rounded-lg">
        <blockquote>{quote}</blockquote>
        <p className="italic font-thin text-right">
          --
          {author}
        </p>
        <button
          type="button"
          className="px-12 py-1 rounded-lg bg-red-300"
          onClick={() => fetchData()}
        >
          Shuffle
        </button>
      </div>
    </div>
  );
};
export default Quotes;
