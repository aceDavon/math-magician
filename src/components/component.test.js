import { fireEvent, render, screen } from '@testing-library/react';
import Welcome from './Welcome';
import Calculator from './Calculator';
import Quotes from './Quotes';

describe(' testing components', () => {
  test('testing landing page', () => {
    const landing = render(<Welcome />);
    expect(landing).toMatchSnapshot();
  });
  test('testing Quote Page', async () => {
    render(<Quotes />);
    const button = await screen.findAllByRole('button');
    expect(button).toHaveLength(1);
  });
  test('testing calculator page', async () => {
    render(<Calculator />);
    const btn = await screen.findAllByRole('button');
    expect(btn).toHaveLength(19);
  });
  it('Simulate user interaction on click of a button', async () => {
    render(<Quotes />);
    const btn = screen.getByRole('button');
    fireEvent.click(btn);
    expect(jest.fn()).not.toBeCalled();
  });
});
