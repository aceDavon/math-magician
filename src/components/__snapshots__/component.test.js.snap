// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` testing components testing landing page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="w-full pt-20 relative h-screen bg-gradient-to-r from-gray-800 to-gray-900 text-white"
      >
        <img
          alt="home-minimalist-bg"
          class="w-full h-screen top-0 object-cover absolute mix-blend-overlay"
          src="landing.webp"
        />
        <h1
          class="text-4xl mt-28 mx-10 text-red-400"
        >
          Welcome to Our Page!
        </h1>
        <p
          class="text-base my-8 mx-10"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque rerum atque optio quaerat consequuntur a facere quasi modi neque maiores veniam et accusamus, porro ducimus exercitationem dignissimos tempora dolore unde perspiciatis non pariatur quia consectetur, velit laudantium. Ut, placeat vero! Aut, natus omnis. Amet, magni? Soluta enim nulla necessitatibus quisquam nam quis, consequatur velit laudantium libero ad alias fuga deserunt amet provident quae ea beatae.
        </p>
        <p
          class="text-base my-5 mx-10"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque rerum atque optio quaerat consequuntur a facere quasi modi neque maiores veniam et accusamus, porro ducimus exercitationem dignissimos tempora dolore unde perspiciatis non pariatur quia consectetur, velit laudantium. Ut, placeat vero! Aut, natus omnis. Amet, magni? Soluta enim nulla necessitatibus quisquam nam quis, consequatur velit laudantium libero ad alias fuga deserunt amet provident quae ea beatae.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="w-full pt-20 relative h-screen bg-gradient-to-r from-gray-800 to-gray-900 text-white"
    >
      <img
        alt="home-minimalist-bg"
        class="w-full h-screen top-0 object-cover absolute mix-blend-overlay"
        src="landing.webp"
      />
      <h1
        class="text-4xl mt-28 mx-10 text-red-400"
      >
        Welcome to Our Page!
      </h1>
      <p
        class="text-base my-8 mx-10"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque rerum atque optio quaerat consequuntur a facere quasi modi neque maiores veniam et accusamus, porro ducimus exercitationem dignissimos tempora dolore unde perspiciatis non pariatur quia consectetur, velit laudantium. Ut, placeat vero! Aut, natus omnis. Amet, magni? Soluta enim nulla necessitatibus quisquam nam quis, consequatur velit laudantium libero ad alias fuga deserunt amet provident quae ea beatae.
      </p>
      <p
        class="text-base my-5 mx-10"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Eaque rerum atque optio quaerat consequuntur a facere quasi modi neque maiores veniam et accusamus, porro ducimus exercitationem dignissimos tempora dolore unde perspiciatis non pariatur quia consectetur, velit laudantium. Ut, placeat vero! Aut, natus omnis. Amet, magni? Soluta enim nulla necessitatibus quisquam nam quis, consequatur velit laudantium libero ad alias fuga deserunt amet provident quae ea beatae.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
